frontend/src/pages/SettingsPage.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { useCheckAuthenticated } from '@semapps/auth-provider';
+import { email, required, useAuthProvider, useNotify, useTranslate, } from 'react-admin';
+import { SimpleForm, TextInput } from 'react-admin';
+import { Container } from '@material-ui/core';
+import HeaderTitle from '../layout/HeaderTitle';
+
+const validateEmail = [required(), email('app.validation.email')]
+const validateConfirmNewPassword = [(value, { newPassword, confirmNewPassword }) => {
+	if (!newPassword) return;
+
+	if (newPassword !== confirmNewPassword) {
+		return 'app.validation.confirmNewPassword'
+	}
+	return;
+}];
+
+
+const SettingsPage = () => {
+	const translate = useTranslate();
+	const notify = useNotify();
+	const { identity } = useCheckAuthenticated();
+	const authProvider = useAuthProvider();
+
+	const [formDefaultValue, setFormDefaultValue] = React.useState({
+		email: "",
+		currentPassword: "",
+		newPassword: "",
+		confirmNewPassword: ""
+	})
+
+	React.useEffect(() => {
+		authProvider.getAccountSettings().then((res) => {
+			setFormDefaultValue({ ...formDefaultValue, email: res.email })
+		});
+		// eslint-disable-next-line
+	}, [setFormDefaultValue, authProvider])
+
+	if (!identity?.id) return null;
+
+	return (
+		<>
+			<HeaderTitle>
+				{translate('app.page.settings')}
+			</HeaderTitle>
+			<Container>
+				<SimpleForm initialValues={formDefaultValue}
+					save={(params) => {
+						authProvider.updateAccountSettings({ ...params }).then(() => {
+							notify('auth.message.account_settings_updated', 'success');
+						}).catch((error) => {
+							notify(error.message, 'error');
+						})
```

Comment:
Generally it's better to use `useCallback` for such case, it helps a bit for the performance (inline functions force a full rerender on each render) and make the code easier to read.

Reply:
This should be fine now. ^^